src/plugins/workspace/server/saved_objects/integration_tests/workspace_saved_objects_client_wrapper.test.ts

```diff
@@ -924,4 +928,68 @@ describe('WorkspaceSavedObjectsClientWrapper', () => {
       expect(SavedObjectsErrorHelpers.isNotFoundError(ACLError)).toBe(true);
     });
   });
+
+  describe('data source', () => {
+    beforeAll(async () => {
+      await repositoryKit.create(
+        internalSavedObjectsRepository,
+        DATA_SOURCE_SAVED_OBJECT_TYPE,
+        {
+          title: 'Global data source',
+        },
+        {
+          id: 'global-data-source',
+        }
+      );
+      await repositoryKit.create(
+        internalSavedObjectsRepository,
+        DATA_SOURCE_SAVED_OBJECT_TYPE,
+        {
+          title: 'Data source in workspace 1',
+        },
+        {
+          id: 'data-source-in-workspace-1',
+          workspaces: ['workspace-1'],
+        }
+      );
+    });
+
+    it('should throw permission error when get global data source with non dashboard admin', async () => {
```

Comment:
What if non-admin user want to get all the data sources they have permission with across all the workspaces they have read permission?

Reply:
For this case, the user will use saved objects client find method. The `validateWorkspacesAndSavedObjectsPermissions` won't be called during the find process. So I think it would be fine.

Reply:
I see, thanks for clarification.